activerecord/lib/active_record/associations/association_scope.rb

```diff
@@ -135,13 +135,15 @@ def add_constraints(scope, owner, chain)
 
               if scope_chain_item == chain_head.scope
                 scope.merge! item.except(:where, :includes, :unscope, :order)
-              elsif !item.references_values.empty?
+              else
                 scope.merge! item.only(:joins, :left_outer_joins)
 
-                associations = item.eager_load_values | item.includes_values
+                if !item.references_values.empty?
```

Comment:
`references_values` is related to `eager_load` and `includes`, not `joins` and `left_outer_joins`.